classes/PaymentModule.php

```diff
@@ -377,7 +377,7 @@ public function validateOrder(
                     $transaction_id = null;
                 }
 
-                if (!$order->addOrderPayment($amount_paid, null, $transaction_id)) {
+                if (!isset($order) || !$order->addOrderPayment($amount_paid, null, $transaction_id)) {
```

Comment:
Is there a relation between this isset and the duplicate order issue ?

Reply:
I don't think so, for the $order to be not defined `$package_list` must not exists which is hardly possible but I guess @PierreRambaud just wanted to make sure ;)